plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogic.java

```diff
@@ -0,0 +1,243 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.plugin.calendar;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.slf4j.Logger;
+
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Reference;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Service;
+
+/**
+ * This class is for the Calendar Authentication and Logic, parts of the Code are from
+ * https://developers.google.com/calendar/quickstart/java
+ * 
+ * @author Patrick Gebhardt, Florian Bauer
+ */
+@Service
+public class CalendarLogic {
+	
+	@Reference
+	private CalendarService calendarService;
+	@Reference 
+	private Logger logger;
+
+	private List<String> eventList = new ArrayList<>();
+
+	/**
+	 * This method lists the next events from the calendar
+	 * 
+	 * @param number
+	 *            number of events the user wants to get
+	 * @return event summary
+	 */
+	public String getEvents(String number) {
+		try {
+			this.eventList.clear();
+			DateTime now = new DateTime(System.currentTimeMillis());
+			Events events = this.calendarService.getService().events().list("primary").setMaxResults(Integer.valueOf(number)).setTimeMin(now)
+					.setOrderBy("startTime").setSingleEvents(true).execute();
+			List<Event> items = events.getItems();
+			if (items.isEmpty()) {
+				return "No upcoming events found.";
+			}
+			for (Event event : items) {
+				DateTime start = event.getStart().getDateTime();
+				eventCalc(start, event, true);
+			}
+			return "You have following upcoming events:\n" + String.join("\n", this.eventList);
+		} catch (IOException e) {
+			this.logger.error("Sorry, I am not able to get your events.", e);
+			return "An error occured.";
+		}
+
+	}
+
+	/**
+	 * This method contains the logic to show the calendar events today and tomorrow
+	 * 
+	 * @param day
+	 * 			true if it is today, false for tomorrow	
+	 * @return the events of the chosen day
+	 */
+	public String getEventsByDay(boolean day) {
+		try {
+			this.eventList.clear();
+			DateTime now = new DateTime(System.currentTimeMillis());
+			Events events = this.calendarService.getService().events().list("primary").setTimeMin(now).setOrderBy("startTime")
+					.setSingleEvents(true).execute();
+			List<Event> items = events.getItems();
+			if (items.isEmpty()) {
+				return "No upcoming events found.";
+			}
+			if(!day) {
+				Date today = new Date();
+				LocalDateTime plusOne = today.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
```

Comment:
why?

Reply:
to differentiate between today and tomorrow, this was the easiest implementation to get the date for the next day I could find. Maybe I should change the name of the variable day or what do you mean?